editoast/src/views/rolling_stocks.rs

```diff
@@ -0,0 +1,275 @@
+use crate::diesel::{QueryDsl, RunQueryDsl};
+use crate::error::Result;
+use crate::tables::osrd_infra_rollingstock;
+use crate::tables::osrd_infra_rollingstock::dsl;
+use crate::DbPool;
+use actix_web::dev::HttpServiceFactory;
+use actix_web::get;
+use actix_web::web::{self, block, Data, Json, Path};
+use diesel::result::Error as DieselError;
+use diesel::PgConnection;
+use editoast_derive::EditoastError;
+use serde::Serialize;
+use serde_json::{json, Map, Value as JsonValue};
+use thiserror::Error;
+
+pub fn routes() -> impl HttpServiceFactory {
+    web::scope("/rolling_stock").service(get)
+}
+
+/// =================================
+/// Endpoints
+/// =================================
+
+/// Return a specific set of electrical profiles
+#[get("/{rolling_stock}")]
+async fn get(db_pool: Data<DbPool>, rolling_stock: Path<i64>) -> Result<Json<RollingStock>> {
+    let rolling_stock = rolling_stock.into_inner();
+    block(move || {
+        let mut conn = db_pool.get().expect("Failed to get DB connection");
+        Ok(Json(RollingStock::retrieve(&mut conn, rolling_stock)?))
+    })
+    .await
+    .unwrap()
```

Comment:
We should block and get a connection as late as possible.
The idea is to use `db_pool` everywhere and when you need to make a request, you block and get a connection.

You will need a `DbPool` for your tests. You can take example here: https://github.com/DGEXSolutions/osrd/blob/ad34f7ce580b1f150abada912f6344ee7450cac2/editoast/src/documents.rs#L154-L155

I know that's not what we're currently doing in most endpoints, but that should be fixed.